### ISSUE TITLE
New install. Virtualmin gpl. IPv6 only server. Error. No map sources found in the Postfix configuration.

### ISSUE THREAD

**Issue by chris001** [webmin/webmin#537]
```
The status of your system is being checked to ensure that all enabled features are available, that the mail server is properly configured, and that quotas are active ..

    Your system has 512 MB of memory, which is at or above the Virtualmin recommended minimum of 256 MB.

    BIND DNS server is installed, and the system is configured to use it.

    A problem was found with your Postfix virtual maps : No map sources were found in the Postfix configuration

.. your system is not ready for use by Virtualmin.
```
Ubuntu 16.04 server. IPv6 only.
Had to `apt install bind9 postfix` manually.
It's as if virtualmin isn't configuring any of the basic defaults for postfix.
Don't recall this error happening on an IPv4 server.

**Comment by chris001** [comment]
When you click, "Create Virtual Server", fill in the fully qualified domain name, choose a username and password, and click "Create Server", this is the error:
```
Failed to create virtual server : No virtual domains file (virtual_alias_maps) was found in your Postfix configuration!
```

**Comment by chris001** [comment]
( The solution to the above errors was to run install.sh.  Apparently, installing the webmin and virtualmin debs directly, although they technically installed and appeared to work, they fail to generate required basic config files needed for the software to actually work. )

Another issue/bug found while trying to get an LE TLS cert, on IPv6 only server, with 20 ports of an IPv4 address, in the high range, well above 10,000.

```
Requesting a certificate for server6.mydomain.com, www.server6.mydomain.com from Let's Encrypt ..
.. request failed : Failed to request certificate :

Parsing account key...
Parsing CSR...
Registering account...
Registered!
Verifying server6.mydomain.com...
Wrote file to /home/server6/public_html/.well-known/acme-challenge/lqaVjbTFr0i0kVZtneclmUKvyEFXiXJnzjLfowye8sU, but couldn't download http://server6.mydomain.com/.well-known/acme-challenge/lqaVjbTFr0i0kVZtneclmUKvyEFXiXJnzjLfowye8sU
Traceback (most recent call last):
  File "/usr/share/webmin/webmin/acme_tiny.py", line 203, in <module>
    main(sys.argv[1:])
  File "/usr/share/webmin/webmin/acme_tiny.py", line 199, in main
    signed_crt = get_crt(args.account_key, args.csr, args.acme_dir, log=LOGGER, CA=args.ca)
  File "/usr/share/webmin/webmin/acme_tiny.py", line 154, in get_crt
    domain, challenge_status))
ValueError: server6.mydomain.com challenge did not pass: {u'status': u'invalid', u'validationRecord': [{u'url': u'http://server6.mydomain.com/.well-known/acme-challenge/lqaVjbTFr0i0kVZtneclmUKvyEFXiXJnzjLfowye8sU', u'hostname': u'server6.mydomain.com', u'addressUsed': u'104.99.99.99', u'port': u'80', u'addressesResolved': [u'104.99.99.99', u'2607:9999:999:eeee::1']}], u'keyAuthorization': u'lqaVjbTFr0i0kVZtneclmUKvyEFXiXJnzjLfowye8sU.FvbbxMG-KV7uvvxuQPtaBhpesTyD5uvX56n0voska38', u'uri': u'https://acme-v01.api.letsencrypt.org/acme/challenge/hpj_2DlSeq6eOJU-QvBnTSZxz5jezaQbysc_qh9LVvY/1084721393', u'token': u'lqaVjbTFr0i0kVZtneclmUKvyEFXiXJnzjLfowye8sU', u'error': {u'status': 403, u'type': u'urn:acme:error:unauthorized', u'detail': u'Invalid response from http://server6.mydomain.com/.well-known/acme-challenge/lqaVjbTFr0i0kVZtneclmUKvyEFXiXJnzjLfowye8sU: "<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>404 Not Found</title>\n</head><body>\n<h1>Not Found</h1>\n<p"'}, u'type': u'http-01'}
```

**Comment by jcameron** [comment]
It looks like the Let's Encrypt service failed to validate your domain. The cause may be that it isn't externally resolvable, or that Let's Encrypt doesn't support IPv6-only websites.

**Comment by jcameron** [comment]
Regarding your original error - the correct solution is to run install.sh, as this sets up a bunch of dependencies that if missing will trigger errors like that.

**Comment by chris001** [comment]
First, my basic assumption, being sensible, is that Let's Encrypt doesn't depend on which IP address type is used, IP4 or IP6, and neither does any modern server software, it uses the hostname requested to route the incoming request to the corresponding virtual server on the shared IP4 or IP6 address. 

The Let's Encrypt issue has to do with the https default port 443 used for the IP4 address.  There either needs to be a way to specify for Let's Encrypt to hit the https server back on a non-standard IP4 port.  The reason is, this virtual private server is a modern mostly IP6 only server.  It's on a shared IP4 with many other servers, has an IP4 with only 20 ports of that IP4 dedicated to itself.  The port range, on this shared IP4, in this instance, happens to be 13800-13820.  So, either, if there is a way to configure the https server on the virtual server to use port 13801, and specify to let's encrypt code, to hit the https server on the domain at port 13801, this let's encrypt verification procedure would work.

So to summarize, the solution to this issue is:
1. Either specify a nonstandard https port and configure the https server to run on that port.  
2. Or, use a hostname which resolves only to the server's IP6 address, on which all ports are available, and ask lets encrypt to validate that hostname, which will resolve to the IP6 address only, and Let's encrypt will validate against that IP6 address on default https port 443.

**Comment by jcameron** [comment]
Unfortunately there is no way I know of for the Let's Encrypt service to be told which port to connect to .... which is really annoying for Webmin/Virtualmin, as it prevents us from easily requesting a cert for Webmin itself using it's own webserver. 

There is a good reason for this though - if Let's Encrypt allowed non-443 ports, a non-root user on a shared system could request a cert for a domain that they don't actually own.

**Comment by chris001** [comment]
I found a workaround, just to remove the IP4 A record from DNS.  Then, the LE script works, it finds on the AAAA record in DNS, and the cert request works, because all ports are available on the IP6 address.

**Comment by jcameron** [comment]
Ok, that makes sense.

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
